contest/register_contest.py

```diff
@@ -19,25 +19,61 @@
     '''
 from contest.models import Contest
 from contest.models import Contestant
-
 from contest.contest_info import get_contest_or_404
+from contest.contest_info import user_can_register_contest
 from contest.contest_info import can_register
-
+from contest import public_user
 from group.models import Group
+from users.models import User
 
-from utils.log_info import get_logger
+from utils.log_info import get_logger 
 
 logger = get_logger()
 
-def register_user(user, contest):
+# add one contestant
+def add_contestant(user, contest):
+    contestant = Contestant(contest = contest, user = user)
+    contestant.save()
+    logger.info('Contest: User %s attends Contest %s!' % (user.username, contest.id))
+
+# add many contestants
+def add_contestants(users, contest):
+    for user in users:
+         add_contestant(user, contest)
+
+# add many contestants and activate them
+def add_contestants_and_activate(users, contest):
+     add_contestants(users, contest)
+     public_user.activate_public_users(users)
+
+def user_register_contest(user, contest):
     if can_register(user, contest):
-        contestant = Contestant(contest = contest, user = user)
-        contestant.save()
-        logger.info('Contest: User %s attends Contest %s!' % (user.username, contest.id))
+        add_contestant(user, contest)
+
+def group_register_contest(group, contest):
```

Comment:
Please check whether the contest start or not. (We can not register a started contest)